src/types/bytes.rs

```diff
@@ -97,6 +97,24 @@ impl PyBytes {
     }
 }
 
+impl Py<PyBytes> {
+    /// Gets the Python bytes as a byte slice. Because Python bytes are
+    /// immutable, the result may be used for as long as the reference to
+    /// `self` is held.
+    pub fn as_bytes(&self, _py: Python<'_>) -> &[u8] {
```

Comment:
I understand that it is implied, but since it seems to be the point of this, wouldn't it be nice to make it explicit which lifetime is attached to the return value, e.g.
```rust
pub fn as_bytes<'a>(&'a self, _py: Python<'_>) -> &'a [u8] {
```

Reply:
Good call, done.

Reply:
> I understand that it is implied, but since it seems to be the point of this, wouldn't it be nice to make it explicit which lifetime is attached to the return value, e.g.

It's not about being explicit - this function is unsound without it. Without `&'a self` this function can produce a slice with an unbounded lifetime.